File: Try_GameEngine5/src/com/example/try_gameengine/avg/LSystem.java
package com.example.try_gameengine.avg;

import java.util.Random;

import android.app.Activity;
import android.content.Context;

public class LSystem {
	public static int DEFAULT_MAX_CACHE_SIZE = 10;
	public static int w =800;
	public static int h = 1200;
	static public String encoding = "UTF-8";
	public static Random random = new Random();
	static SystemHandler systemHandler;
	final static public String FONT_NAME = "Monospaced";
	final static public int DEFAULT_MAX_FPS = 50;
	
	public static void gc(){
		System.gc();
	}
	
	public static int getRandom(int i, int j) {
		Random random = new Random();
		return i + random.nextInt((j - i) + 1);
	}

	public static SystemHandler getSystemHandler() {
		return systemHandler;
	}

	public static void setSystemHandler(SystemHandler systemHandler) {
		LSystem.systemHandler = systemHandler;
	}
	
	
	
//	public static int getRandom() {
//		Random random = new Random();
//		return i + random.nextInt((j - i) + 1);
//	}
}

class SystemHandler{
	Activity activity;

	public Activity getActivity() {
		return activity;
	}

	public void setActivity(Activity activity) {
		this.activity = activity;
	}
	
	public Context getContext(){
		return activity;
	}
	
	private AssetsSoundManager asm;
	
	/**
	 * 
	 * @return
	 */
	public AssetsSoundManager getAssetsSound() {
		if (this.asm == null) {
			this.asm = AssetsSoundManager.getInstance(activity);
		}
		return asm;
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/avg/LSystem.kt
package com.example.try_gameengine.avg

import android.app.Activity
import android.content.Context
import java.util.Random

object LSystem {
    var DEFAULT_MAX_CACHE_SIZE: Int = 10
    var w: Int = 800
    var h: Int = 1200
    var encoding: String = "UTF-8"
    var random: Random = Random()

    //	public static int getRandom() {
    var systemHandler: SystemHandler? = null
    const val FONT_NAME: String = "Monospaced"
    const val DEFAULT_MAX_FPS: Int = 50

    fun gc() {
        System.gc()
    }

    fun getRandom(i: Int, j: Int): Int {
        val random = Random()
        return i + random.nextInt((j - i) + 1)
    }

    //		Random random = new Random();
    //		return i + random.nextInt((j - i) + 1);
    //	}
}

class SystemHandler {
    var activity: Activity? = null

    val context: Context?
        get() = activity

    private var asm: AssetsSoundManager? = null

    val assetsSound: AssetsSoundManager?
        /**
         * 
         * @return
         // */
        get() {
            if (this.asm == null) {
                this.asm = AssetsSoundManager.Companion.getInstance(activity)
            }
            return asm
        }
}
